packages/vue-app/template/store.js

```diff
@@ -3,130 +3,136 @@ import Vuex from 'vuex'
 
 Vue.use(Vuex)
 
-let storeData = {}
-
-let files
+const VUEX_PROPERTIES = ['state', 'getters', 'actions', 'mutations']
+let store = {}
+let fileResolver
 
 void (function updateModules() {
-  files = require.context('@/<%= dir.store %>', true, /^\.\/(?!<%= ignorePrefix %>)[^.]+\.(<%= extensions %>)$/)
-  const filenames = files.keys()
+  fileResolver = require.context('@/<%= dir.store %>', true, /^\.\/(?!<%= ignorePrefix %>)[^.]+\.(<%= extensions %>)$/)
+
+  // Paths are sorted from low to high priority (for overwriting properties)
+  const paths = fileResolver.keys().sort((p1, p2) => {
+    let res = p1.split('/').length - p2.split('/').length
+
+    if (res === 0) {
+      if (p1.includes('/index.')) res = -1
+      if (p2.includes('/index.')) res = 1
+    }
+    return res
+  })
 
   // Check if {dir.store}/index.js exists
-  const indexFilename = filenames.find(filename => filename.includes('./index.'))
+  const indexPath = paths.find(path => path.includes('./index.'))
 
-  if (indexFilename) {
-    storeData = getModule(indexFilename)
+  if (indexPath) {
+    store = requireModule(indexPath, { isRoot: true })
+  }
+
+  // If store is an exported method = classic mode (deprecated)
+  if (typeof store === 'function') {
+    const log = (process.server ? require('consola') : console)
+    return log.warn('Classic mode for store/ is deprecated and will be removed in Nuxt 3.')
   }
 
-  // If store is not an exported method = modules store
-  if (typeof storeData !== 'function') {
-    // Store modules
-    if (!storeData.modules) {
-      storeData.modules = {}
+  // Enforce store modules
+  store.modules = store.modules || {}
+
+  for (const path of paths) {
+    // Remove store path + extension (./foo/index.js -> foo/index)
+    const namespace = path.replace(/^\.\//, '').replace(/\.(<%= extensions %>)$/, '')
+
+    // Ignore indexFile, handled before
+    if (namespace === 'index') continue
```

Comment:
Same here as @galvez mentioned before. 

We should prefer `if (cond){ doSthHere }` instead of `if (cond) doSth`. :relaxed: